### PR TITLE
KOGITO-1196: Editor goes back horizontally when diagram is bigger than default

### PR DESCRIPTION
**Pull request by handreyrc**
@romartin, @inodeman: Could you review this PR please?

Assembly with https://github.com/kiegroup/lienzo-tests/pull/92

### CODE REVIEW

**Comment by romartin** [comment]
Can you maybe @hasys also do a quick review here plz?

**Comment by handreyrc** [comment]
@romartin I updated the code as we agreed. The MouseDragSynchronization will be disabled by default. However, it can be enabled/disabled whenever needed.
Could you review this PR again?
Related PR
[lienzo-tests PR#92](https://github.com/kiegroup/lienzo-tests/pull/92)

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Full downstream build failing due to unrelated test fail. I suggest not retrigger it again and finish the review, since binaries are generated. Thank you!

<details>
  <summary>Failed test</summary>
  <pre>org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36511/kie-server-services/services/rest/server'}]

Error Message

Synchronization failed for defined timeout: 240000 milliseconds.

Stacktrace

java.util.concurrent.TimeoutException: Synchronization failed for defined timeout: 240000 milliseconds.
	at org.kie.server.integrationtests.controller.WebSocketKieControllerStartupIntegrationTest.testContainerDisposedAfterStartup(WebSocketKieControllerStartupIntegrationTest.java:248)

Standard Output

05:51:39.545 [main] INFO  o.k.s.i.c.WebSocketKieControllerStartupIntegrationTest - ********** Test name testContainerDisposedAfterStartup[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:36511/kie-server-services/services/rest/server'}] ***********
05:51:49.560 [main] WARN  o.k.s.s.i.ControllerBasedStartupStrategy - Unable to connect to any controllers, delaying container installation until connection can be established
05:51:49.990 [Acme.Utils.ThreadPool(33)-PooledThread: Acme.Serve.Serve$ServeConnection@61f34cbd] ERROR o.k.s.services.impl.KieServerImpl - Error creating container 'kie-concurrent' for module 'org.kie.server.testing:stateless-session-kjar:1.0.0'
java.lang.IllegalStateException: There's already another KieContainer created with the id kie-concurrent
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:211)
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:176)
	at org.kie.server.services.impl.KieServerImpl.createContainer(KieServerImpl.java:281)
	at org.kie.server.remote.rest.common.resource.KieServerRestImpl.createContainer(KieServerRestImpl.java:157)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:140)
	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:509)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:399)
	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$0(ResourceMethodInvoker.java:363)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:365)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:337)
	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:310)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:443)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:233)
	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:139)
	at org.jboss.resteasy.core.interception.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:358)
	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:142)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:219)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:227)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
	at org.jboss.resteasy.plugins.server.tjws.TJWSServletDispatcher.service(TJWSServletDispatcher.java:44)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at Acme.Serve.Serve$ServeConnection.runServlet(Serve.java:2331)
	at Acme.Serve.Serve$ServeConnection.parseRequest(Serve.java:2285)
	at Acme.Serve.Serve$ServeConnection.run(Serve.java:2057)
	at Acme.Utils$ThreadPool$PooledThread.run(Utils.java:1406)
	at java.lang.Thread.run(Thread.java:748)
05:52:09.792 [KieServer-ControllerConnect] ERROR o.k.s.services.impl.KieServerImpl - Error creating container 'kie-concurrent' for module 'org.kie.server.testing:stateless-session-kjar:1.0.0'
java.lang.IllegalStateException: There's already another KieContainer created with the id kie-concurrent
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:211)
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:176)
	at org.kie.server.services.impl.KieServerImpl.createContainer(KieServerImpl.java:281)
	at org.kie.server.services.impl.ContainerManager.installContainersSync(ContainerManager.java:44)
	at org.kie.server.services.impl.ContainerManager.installContainers(ContainerManager.java:33)
	at org.kie.server.services.impl.controller.ControllerConnectRunnable.run(ControllerConnectRunnable.java:73)
	at java.lang.Thread.run(Thread.java:748)
05:55:50.437 [XNIO-1 task-2] ERROR o.k.s.services.impl.KieServerImpl - Error creating container 'kie-concurrent' for module 'org.kie.server.testing:stateless-session-kjar:1.0.0'
java.lang.IllegalStateException: There's already another KieContainer created with the id kie-concurrent
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:211)
	at org.drools.compiler.kie.builder.impl.KieServicesImpl.newKieContainer(KieServicesImpl.java:176)
	at org.kie.server.services.impl.KieServerImpl.createContainer(KieServerImpl.java:281)
	at org.kie.server.services.impl.KieContainerCommandServiceImpl.executeScript(KieContainerCommandServiceImpl.java:147)
	at org.kie.server.controller.websocket.client.handlers.CommandScriptMessageHandler.onMessage(CommandScriptMessageHandler.java:68)
	at org.kie.server.controller.websocket.common.handlers.KieServerMessageHandler.onMessage(KieServerMessageHandler.java:48)
	at org.kie.server.controller.websocket.common.handlers.KieServerMessageHandler.onMessage(KieServerMessageHandler.java:27)
	at io.undertow.websockets.jsr.FrameHandler$7.run(FrameHandler.java:286)
	at io.undertow.websockets.jsr.ServerWebSocketContainer$1.call(ServerWebSocketContainer.java:170)
	at io.undertow.websockets.jsr.ServerWebSocketContainer$1.call(ServerWebSocketContainer.java:167)
	at io.undertow.websockets.jsr.ServerWebSocketContainer.invokeEndpointMethod(ServerWebSocketContainer.java:604)
	at io.undertow.websockets.jsr.ServerWebSocketContainer$6.run(ServerWebSocketContainer.java:590)
	at io.undertow.websockets.jsr.OrderedExecutor$ExecutorTask.run(OrderedExecutor.java:67)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1871)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1451)
	at java.lang.Thread.run(Thread.java:748)

Standard Error

Apr 30, 2020 5:51:39 AM Acme.Serve.Serve log
INFO: Servlet for path '/' already defined and no default will be used.
Apr 30, 2020 5:59:50 AM Acme.Serve.Serve log
INFO: Accept: java.net.SocketException: Socket closed

</pre>
</details>

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by manstis** [comment]
I was not able to build `kie-wb-common` (latest fetched from `master`) due to:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project kie-wb-common-stunner-bpmn-marshalling: Compilation failure
[ERROR] /home/manstis/workspaces/git/manstis/forks/kie-wb-common/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-marshalling/src/main/java/org/kie/workbench/common/stunner/bpmn/client/marshall/converters/tostunner/FlowElementConverter.java:[61,32] incompatible types: invalid method reference
[ERROR]     incompatible types: java.lang.Object cannot be converted to org.eclipse.bpmn2.Gateway
```
Do I need to compile something else somewhere too?

**Comment by handreyrc** [comment]
Hey @manstis, You don't need do build anything else. A rebase should do the job.

**Comment by handreyrc** [comment]
@manstis I found out the snapshot version was updated to 7.38.0-SNAPSHOT just in case you are testing lienzo-core with stunner. The changes won't take any effect if you don't update the pom file.

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by manstis** [comment]
Is it just me... or has the content of this PR changed completely!? It used to enable/disable the mediator but now it has been deleted entirely. I also note the FDB now fails too:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project kie-wb-common-stunner-lienzo: Compilation failure: Compilation failure: 
[ERROR] /home/jenkins/workspace/KIE/master/upstream-downstream/lienzo-core.downstream/kiegroup_kie-wb-common/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/components/mediators/LienzoCanvasMediators.java:[118,24] cannot find symbol
[ERROR]   symbol:   method isMouseDragSynchronizationEnabled()
[ERROR]   location: variable handler of type com.ait.lienzo.client.widget.panel.scrollbars.ScrollablePanelHandler
[ERROR] /home/jenkins/workspace/KIE/master/upstream-downstream/lienzo-core.downstream/kiegroup_kie-wb-common/kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/main/java/org/kie/workbench/common/stunner/client/lienzo/components/mediators/LienzoCanvasMediators.java:[119,24] cannot find symbol
[ERROR]   symbol:   method disableMouseDragSynchronization()
[ERROR]   location: variable handler of type com.ait.lienzo.client.widget.panel.scrollbars.ScrollablePanelHandler
```
Please let me know when this is OK to review (I want to run Business Central to test it)

Perhaps everybody needs to re-review it now?

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
@handreyrc, @manstis, @mareknovotny,

As I can see regular Full downstream build was replaced by new one, any chance it is not supporting Lienzo-core yet? I mean maybe it is not building lienzo-core from this repo and starts directly with kie-wb-common. Is it possible?

**Comment by mareknovotny** [comment]
not sure what do you mean by that?

> @handreyrc, @manstis, @mareknovotny,
> 
> As I can see regular Full downstream build was replaced by new one, any chance it is not supporting Lienzo-core yet? I mean maybe it is not building lienzo-core from this repo and starts directly with kie-wb-common. Is it possible?


why FDB should jump directly to kie-wb-common? It is setup and it was triggered as you could see in https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/upstream-downstream/job/lienzo-core.downstream/9/

**Comment by hasys** [comment]
@mareknovotny, yes it's true.

@handreyrc I guess I found the issue. See: 

<details>
  <summary>kie-wb-common cloned repo</summary>
  <pre>
Finished examining handreyrc/kie-wb-common

Found KOGITO-1196 at revision dd779f04e4b6c29803b994f854ddeff31a811d36
[Pipeline] checkout
Cloning repository https://github.com/handreyrc/kie-wb-common.git
 > git init /home/jenkins/workspace/KIE/master/upstream-downstream/lienzo-core.downstream/kiegroup_kie-wb-common # timeout=10
Fetching upstream changes from https://github.com/handreyrc/kie-wb-common.git
 > git fetch --no-tags --progress https://github.com/handreyrc/kie-wb-common.git +refs/heads/KOGITO-1196:refs/remotes/origin/KOGITO-1196 # timeout=10
Fetching without tags
Checking out Revision dd779f04e4b6c29803b994f854ddeff31a811d36 (KOGITO-1196)
 > git config remote.origin.url https://github.com/handreyrc/kie-wb-common.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/KOGITO-1196:refs/remotes/origin/KOGITO-1196 # timeout=10
 > git config remote.origin.url https://github.com/handreyrc/kie-wb-common.git # timeout=10
 > git fetch --no-tags --progress https://github.com/handreyrc/kie-wb-common.git +refs/heads/KOGITO-1196:refs/remotes/origin/KOGITO-1196 # timeout=10
 > git config core.sparsecheckout # timeout=10
 > git checkout -f dd779f04e4b6c29803b994f854ddeff31a811d36 # timeout=10
Commit message: "KOGITO-1196: Editor goes back horizontally when diagram is bigger than default"</pre>
</details>

in other words, Jenkins building kie-wb-common not from the master but from your other PR: https://github.com/kiegroup/kie-wb-common/pull/3268

It because it has the same branch name as in this repo. Our jenkins when building Full Downstream Build building all branches with same name/author if any instead of master.

You need to not only close the kie-wb-comon PR, but also remove branch from the github to to fix this build.

**Comment by handreyrc** [comment]
@hasys  Good catch! I'll do it. Thanks

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Hi @manstis,

at least from BPMN Designer point of view described behaviour is the same as on master and was introduced with "infinite canvas" some time ago.

**Comment by manstis** [comment]
@hasys FYI, the Guided Decision Table editor uses the same infinite canvas (that is where it was first invented) and does not suffer the problem of both axis extending... IDK if anybody wants to log a JIRA for it in Stunner. I'll leave that up to those closer to the code; as there may be one already.

**Comment by hasys** [comment]
@manstis I am for infinite canvas in both directions, when you can scroll as long as you want, so I don't see an issue in addition free space. I think this is not reported yet, if you think it is an issue and should be fixed feel free to report an issue. Thank you!
